landlab/components/species_evolution/species_evolver.py

```diff
@@ -0,0 +1,673 @@
+    array([[ 1.,  1.,  1.,  1.,  1.,  1.,  1.],
+           [ 1.,  1.,  1.,  1.,  1.,  1.,  1.],
+           [ 1.,  1.,  1.,  1.,  1.,  1.,  1.]])
+
+    Instantiate the component with the grid.
+
+    >>> se = SpeciesEvolver(mg)
+
+    ZoneController requires a function that returns a mask of the total extent
+    of taxa habitat. The mask is a boolean array where `True` values represent
+    nodes that satisfy habitat conditions. Zone objects are not created here.
+    The mask only maps the extent where taxa can exist. This function returns
+    `True` where elevation is below 100, which is where the simulated lowland
+    taxa of this model can inhabit.
+
+    >>> def zone_func(grid):
+    ...     return grid.at_node['topographic__elevation'] < 100
+
+    Instantiate ZoneController with the grid and zone function. The initial
+    zones are created at controller instantiation. In this example, one zone is
+    created because all nodes of the zone mask are adjacent to each other.
+
+    >>> zc = ZoneController(mg, zone_func)
+    >>> len(zc.zones) == 1
+    True
+
+    Additional examples of controller usage are provided in ZoneController
+    documentation.
+
+    The ``mask`` of the zone is True where the conditions of the zone function
+    are met. All nodes of the grid are included because the elevation of each
+    node is below 100.
+
+    >>> zc.zones[0].mask
+    array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True], dtype=bool)
+
+    Populate a taxon to the zone. The attribute, ``taxa_data_frame`` indicates
+    only the one taxon exists because we populated each zone with one taxon,
+    and only one zone exists.
+
+    >>> taxon = zc.populate_zones_uniformly(1)
+    >>> se.track_taxa(taxon)
+    >>> se.taxa_data_frame
+         appeared  latest_time  extant
+    uid
+    0           0            0    True
+
+    Force a change in the zone mask to demonstrate component functionality.
+    Here we begin a new time step where topography is uplifted by 200 that
+    forms a ridge trending north-south in the center of the grid.
+
+    >>> z[[3, 10, 17]] = 200
+    >>> z.reshape(mg.shape)
+    array([[   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.]])
+
+    The current elevation, the elevation following uplift, is represented here.
+
+    - - - ^ - - -       elevation:  - 1
+    - - - ^ - - -                   ^ 200
+    - - - ^ - - -
+
+    The updated zone mask is below.
+
+    . . . x . . .       key:    . node in zone mask
+    . . . x . . .               x node outside of zone mask
+    . . . x . . .
+
+    Run a step of both the ZoneController and SpeciesEvolver. Two zones exist
+    following this time step.
+
+    >>> delta_time = 1000
+    >>> zc.run_one_step(delta_time)
```

Comment:
Why do you have to run line 159 to get that there are two zones? Isn't that completely taken care of in line 158?

Reply:
I have considered moving the SpeciesEvolver call down to focus this part of the example on zones. Although, I want to always demonstrate these two running directly together to keep time in sync. So I split it up this way:

```python
    Run a step of both the ZoneController and SpeciesEvolver.
    >>> delta_time = 1000
    >>> zc.run_one_step(delta_time)
    >>> se.run_one_step(delta_time)
    
    Two zones exist following this time step.
    
    >>> len(zc.zones) == 2
    True
```

Reply:
I see. Basically you would never want to run ZoneController without immediately running SpeciesEvolver, right? That makes sense now that you say it, but I appreciated you saying it. What if you said that explicitly? Something like:
Two zone exist following the `run_one_step` call to the ZoneController.
SpeciesEvolver should always be called immediately after ZoneController, so that zones and species do not get out-of-sync.

Maybe that's not exactly right, but you see where I'm going.